presto-main/src/main/java/io/prestosql/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -82,19 +86,32 @@
 
     private final SplitManager splitManager;
     private final Metadata metadata;
+    private final DynamicFilterService dynamicFilterService;
 
     @Inject
-    public DistributedExecutionPlanner(SplitManager splitManager, Metadata metadata)
+    public DistributedExecutionPlanner(SplitManager splitManager, Metadata metadata, DynamicFilterService dynamicFilterService)
     {
         this.splitManager = requireNonNull(splitManager, "splitManager is null");
         this.metadata = requireNonNull(metadata, "metadata is null");
+        this.dynamicFilterService = requireNonNull(dynamicFilterService, "dynamicFilterService is null");
     }
 
     public StageExecutionPlan plan(SubPlan root, Session session)
     {
         ImmutableList.Builder<SplitSource> allSplitSources = ImmutableList.builder();
         try {
-            return doPlan(root, session, allSplitSources);
+            // extract mapping of dynamic filter id to it's join node id
+            ImmutableMap.Builder<String, PlanNodeId> dynamicFilterSourcesBuilder = ImmutableMap.builder();
```

Comment:
do we need this mapping at all? dynamic filter service just cares about `dynamic filter id` and not about `join plan node ids`

Reply:
Right now each `DynamicFilterSourceOperator` builds a single `TupleDomain<String>` even if it is collecting multiple dynamic filters. So when we report the summary to coordinator we used the join node id there. This resulted in the need for this mapping when try to get the summary for a given dynamic filter. To avoid this mapping, I think the summary reporting logic would have to be changed to send `Map<filter id, TupleDomain>`.

Reply:
Removed this mapping in updated changes